test/e2e/validation_test.go

```diff
@@ -53,6 +54,53 @@ type winService struct {
 	description string
 }
 
+// testNodesBecomeReady tests that all Windows nodes become ready and schedulable at least once.
+func (tc *testContext) testNodesBecomeReady(t *testing.T) {
+	nodes := gc.allNodes()
+	for index, node := range nodes {
+		t.Run(node.GetName(), func(t *testing.T) {
+			err := wait.PollImmediate(retry.Interval, retry.ResourceChangeTimeout, func() (done bool, err error) {
+				// Get current node state, and update it in the cache
```

Comment:
What cache are you updating? Isn't nodes a local variable?

If you are updating the cache it shouldn't be a side effect of testing if nodes are ready.

Reply:
gc.Nodes is not a local variable. I agree that it shouldn't be a side effect of testing, but thats already the case with waitForConfiguredWindowsNodes. A large portion of the tests depend on there being an updated node cache available to them via gc, which is why waitForConfiguredNodes have been called at various points in the suite.

I will remove the caching, I don't think it will cause any issues. Being this defensive was an artifact of the cancelled work to move cordoning into WICD.